File: Karaoke/Arquivo-em-Java/controller/FilaController.java
import java.util.Queue;
import java.util.LinkedList;

public class FilaController {
    private FilaKaraoke filaKaraoke;

    public FilaController() {
        this.filaKaraoke = new FilaKaraoke();
    }

    public void adicionarPessoa(String nome, String musica) {
        Pessoa p = new Pessoa(nome, musica);
        filaKaraoke.adicionarPessoa(p);
        System.out.println("Pessoa adicionada à fila!");
    }

    public void removerPessoa(int id) {
        boolean removido = filaKaraoke.removerPessoa(id);
        if (removido) {
            System.out.println("Pessoa removida da fila.");
        } else {
            System.out.println("ID não encontrado.");
        }
    }

    public void alterarMusica(int id, String novaMusica) {
        boolean alterado = filaKaraoke.alterarMusica(id, novaMusica);
        if (alterado) {
            System.out.println("Música alterada com sucesso!");
        } else {
            System.out.println("ID não encontrado.");
        }
    }

    public void chamarProxima() {
        Pessoa p = filaKaraoke.chamarProxima();
        if (p != null) {
            System.out.println("Chamando agora: " + p.getNome() + " para cantar '" + p.getMusica() + "'");
        } else {
            System.out.println("A fila está vazia!");
        }
    }

    public void mostrarFila() {
        if (filaKaraoke.isEmpty()) {
            System.out.println("Fila vazia!");
        } else {
            System.out.println("\n--- A fila está organizada da seguinte maneira: ---");
            for (Pessoa p : filaKaraoke.getFila()) {
                System.out.println(p);
            }
        }
    }
}

File: Karaoke/Arquivo-em-Python/Controller/fila_controller.py
from Model.fila_karaoke import FilaKaraoke
from Model.pessoa import Pessoa

class FilaController:
    def __init__(self):
        self.fila_karaoke = FilaKaraoke()

    def adicionar_pessoa(self, nome, musica):
        p = Pessoa(nome, musica)
        self.fila_karaoke.adicionar_pessoa(p)
        print("Pessoa adicionada à fila!")

    def remover_pessoa(self, id_busca):
        removido = self.fila_karaoke.remover_pessoa(id_busca)
        if removido:
            print("Pessoa removida da fila.")
        else:
            print("ID não encontrado.")

    def alterar_musica(self, id_busca, nova_musica):
        alterado = self.fila_karaoke.alterar_musica(id_busca, nova_musica)
        if alterado:
            print("Música alterada com sucesso!")
        else:
            print("ID não encontrado")

    def chamar_proxima(self):
        p = self.fila_karaoke.chamar_proxima()
        if p:
            print(f"Chamando agora: {p.get_nome()} para cantar '{p.get_musica()}'")
        else:
            print("A fila está vazia.")

    def mostrar_fila(self):
        if self.fila_karaoke.is_empty():
            print("Fila vazia")
        else: 
            print("\nA fila está organizada da seguinte maneira:")
            for p in self.fila_karaoke.get_fila():
                print(p)